### PR TITLE
Fix: Fixed issue where confirm to delete would show even with the setting turned off

### LINKED ISSUE #9012: Bug: Confirmation to delete files shows in recycle bin even when setting is off
### Description

Not sure if it the case for all permanent delete operations but I think it would be better to follow the settings or have it as a separate option.
image

### Steps To Reproduce

1. Turn off "Show a confirmation dialog when deleting files or folders" at Files settings > Preferences > Files and folder
2. Delete content in Recycle bin
3. See dialog

### Expected behavior

Dialog should not show

### Files Version

4.1.44.0

### Windows Version

10.0.19043.1645

### Relevant Assets/Logs
[debug_fulltrust.log](https://github.com/files-community/Files/files/8567226/debug_fulltrust.log)
[debug.log](https://github.com/files-community/Files/files/8567227/debug.log)

### PR DESCRIPTION
**Pull request by ferrariofilippo**
…etting is off

**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9012 

**Details**
I changed the logic so that when the setting is off it never shows the dialog. I think this is the correct behaviour

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ public async Task<ReturnResult> DeleteItemsAsync(IEnumerable<IStorageItemWithPat
 			var deleteFromRecycleBin = source.Select(item => item.Path).Any(path => recycleBinHelpers.IsPathUnderRecycleBin(path));
 			var canBeSentToBin = !deleteFromRecycleBin && await recycleBinHelpers.HasRecycleBin(source.FirstOrDefault()?.Path);
 
-			if (showDialog && ((!permanently && !canBeSentToBin) || UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog)) // Check if the setting to show a confirmation dialog is on
+			if (showDialog && (!permanently && !canBeSentToBin) && UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog) // Check if the setting to show a confirmation dialog is on
```
Comment (gave92) [inline src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs:119]: if (showDialog && ((!permanently && !canBeSentToBin && !deleteFromRecycleBin) || UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog)) // Check if the setting to show a confirmation dialog is on

Comment (gave92) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs]: We only want to change the behaviour when deleting from bin, not in all cases. Pinging @yair100 for confirmation.

**Review by gave92 — DISMISSED** [review]

File: src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs]: Behavior should be consistent imo. Users turning the option off don't expect the prompt to show.

Comment (gave92) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs]: So also when deleting from e.g a network folder (where the file gets permanently deleted without going to bin)?
I'm not 100% sure that Files behaves like explorer in all cases regarding permanent deletion vs sending to bin.

Comment (yaira2) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs]: I'm not too concerned with matching File Explorer, the main thing is to follow the setting that users pick in Files.

Comment (gave92) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs]: As u wish, I just don't want people complaining "I lost my files b/c I thought it was going to bin but Files deleted it" xD

@ferrariofilippo the initial code was the correct one, sorry.

Comment (yaira2) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs]: The default setting is to prompt users

Comment (gave92) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs]: Problem is if the user turns the setting off expecting we have the same behaviour of explorer when deleting files to bin vs permanent. Perhaps we have, I dont' know.

**Review by gave92 — COMMENTED** [review]
Aaah wait this is still wrong, if the user has the setting _on_ tbe dialog should always show. Maybe:

`if (showDialog && UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog) // Check if the setting to show a confirmation dialog is on`

**Comment by jiejasonliu** [comment]
> Aaah wait this is still wrong, if the user has the setting _on_ the dialog should always show. Maybe:
> 
> `if (showDialog && UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog) // Check if the setting to show a confirmation dialog is on`

+1, if we're going the route of disabling the confirmation when it's off, we can just invert the logic entirely; the recycle bin variables aren't relevant anymore.

File: src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs:119]: We can mointor feedback and see if it's an issue

**Comment by gave92** [comment]
I've fixed an issue where deleting a file from network folder did not work if the dialog was turned off. Aside from that, LGTM.
Probably better if this PR gets tested by someone else as well.

File: src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs [hunk 2]
```diff
@@ -147,13 +147,15 @@ public async Task<ReturnResult> DeleteItemsAsync(IEnumerable<IStorageItemWithPat
...
-				{
 					return ReturnResult.Cancelled; // Return if the result isn't delete
-				}
 
 				// Delete selected items if the result is Yes
 				permanently = dialogViewModel.DeletePermanently;
 			}
+			else
+			{
+				permanently |= !canBeSentToBin; // delete permanently if recycle bin is not suppeorted
```
Comment (QuaintMako) [inline src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs:157]: Typo
```suggestion
				permanently |= !canBeSentToBin; // delete permanently if recycle bin is not supported
```

**Review by QuaintMako — APPROVED** [review]
LGTM, works fine on my machine.

**Pull request merged**